sails.io.js

```diff
@@ -661,8 +661,30 @@
     };
 
     /**
+     * Adds an `event` listener that will be invoked a single
+     * time then automatically removed.
+     *
+     * @param  {String}   evName [event name]
+     * @param  {Function} fn     [event handler function]
+     * @return {SailsSocket}
+     */
+
+    SailsSocket.prototype.once = function(event, fn){
+      var self = this;
+
+      function on() {
+        self.off(event, on);
+        fn.apply(this, arguments);
+      }
+
+      on.fn = fn;
```

Comment:
Just stores a reference to the function that the handler will call. I took this method (with small modifications) from the socket.io-client module.